lib/util/result.ts

```diff
@@ -0,0 +1,90 @@
+interface Ok<T> {
+  ok: true;
+  value: T;
+}
+
+interface Err {
+  ok: false;
+  error: Error;
+}
+
+type Res<T> = Ok<T> | Err;
```

Comment:
@zharinov can you please export these types in a new PR, as they are used on public API of Result class.

Reply:
What can go wrong if we don't do this? I'm quite certain these types are kind of private and I designed `Result` class to be the only API available here, but what downsides are?